app/src/main/java/se/sigmaconnectivity/blescanner/ui/help/HelpFragment.kt

```diff
@@ -1,40 +1,65 @@
 package se.sigmaconnectivity.blescanner.ui.help
 
+import android.graphics.Bitmap
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.databinding.DataBindingUtil
 import androidx.fragment.app.Fragment
-import com.google.zxing.BarcodeFormat
-import com.journeyapps.barcodescanner.BarcodeEncoder
 import kotlinx.android.synthetic.main.fragment_help.*
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import se.sigmaconnectivity.blescanner.R
-import se.sigmaconnectivity.blescanner.livedata.observe
+import se.sigmaconnectivity.blescanner.databinding.FragmentHelpBinding
+import se.sigmaconnectivity.blescanner.ui.common.livedata.observe
+
 
 class HelpFragment : Fragment() {
 
     private val vm: HelpViewModel by viewModel()
+    private lateinit var binding: FragmentHelpBinding
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
+        binding = DataBindingUtil.inflate(
+            inflater,
+            R.layout.fragment_help,
+            container,
+            false
+        )
+        binding.vm = vm
+        binding.lifecycleOwner = this
 
         initView()
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_help, container, false)
+
+        return binding.root
     }
 
     private fun initView() {
         vm.userId.observe(this, ::updateUserId)
     }
 
-    private fun updateUserId(userId: String) {
-        val barcodeEncoder = BarcodeEncoder()
-        val bitmap = barcodeEncoder.encodeBitmap(userId, BarcodeFormat.QR_CODE, 500, 500)
-        ivQrCode.setImageBitmap(bitmap)
-        tvQrContent.text = userId
+    private fun updateUserId(bitmap: Bitmap) {
+        ivQrCode.setImageBitmap(replaceWhiteWithTransparent(bitmap))
     }
 
+    //TODO move to use case
```

Comment:
It may be hard, as Bitmap is in Android API, and our UseCases don't depend on it.